proto-public/pbmesh/v2beta1/gateway_class.proto

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v2beta1;
+
+import "pbresource/annotations.proto";
+
+// NOTE: this should align to the GAMMA/gateway-api version, or at least be
+// easily translatable.
+//
+// https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.GatewayClass
+//
+// This is a Resource type.
+message GatewayClass {
+  option (hashicorp.consul.resource.spec) = {scope: SCOPE_CLUSTER};
+
+  // ControllerName is the name of the controller that manages Gateways of this class
+  string controller_name = 1;
+
+  // ParametersRef is a reference to a resource that contains the configuration
+  // parameters corresponding to the GatewayClass.
+  ParametersReference parameters_ref = 2;
+
+  // Description of GatewayClass
+  string description = 3;
+}
+
+// NOTE: this should align to the GAMMA/gateway-api version, or at least be
+// easily translatable.
+//
+// ParametersReference specifies a resource that contains controller-specific configuration
+// for a resource
```

Comment:
Again, Kubernetes controller, just to steer as far away from confusion as humanly possible.

Reply:
Done